app/services/repository/BaseRepository.java

```diff
@@ -157,131 +151,151 @@ public void addResources(@Nonnull List<Resource> aResources, Map<String, String>
    *          The resources to flatten and import
    * @throws IOException
    */
-  public void importResources(@Nonnull List<Resource> aResources, Map<String, String> aMetadata) throws IOException {
-
-    Commit.Diff diff = mTriplestoreRepository.getDiffs(aResources);
+  public void importItems(@Nonnull List<Resource> aResources, Map<String, Object> aMetadata) throws IOException {
 
-    TripleCommit.Header header = new TripleCommit.Header(aMetadata.get(TripleCommit.Header.AUTHOR_HEADER),
-      ZonedDateTime.parse(aMetadata.get(TripleCommit.Header.DATE_HEADER)));
-    mTriplestoreRepository.commit(new TripleCommit(header, diff));
+    Commit.Diff diff = mTriplestoreRepo.getDiffs(aResources);
+    TripleCommit.Header header = getTripleCommitHeaderFromMetadata(aMetadata);
+    mTriplestoreRepo.commit(new TripleCommit(header, diff));
     index(diff);
 
   }
 
   @Override
-  public ResourceList query(@Nonnull String aQueryString, int aFrom, int aSize, String aSortOrder,
-                            Map<String, List<String>> aFilters) {
-    return query(aQueryString, aFrom, aSize, aSortOrder, aFilters, null);
+  public ModelCommonList query(@Nonnull String aQueryString, int aFrom, int aSize, String aSortOrder,
+                               Map<String, List<String>> aFilters, final String... aIndices) {
+    return query(aQueryString, aFrom, aSize, aSortOrder, aFilters, null, checkIndices(aIndices));
   }
 
-  public ResourceList query(@Nonnull String aQueryString, int aFrom, int aSize, String aSortOrder,
-                            Map<String, List<String>> aFilters, QueryContext aQueryContext) {
-    ResourceList resourceList;
+  public ModelCommonList query(@Nonnull String aQueryString, int aFrom, int aSize, String aSortOrder,
+                            Map<String, List<String>> aFilters, QueryContext aQueryContext,
+                            final String... aIndices) {
+    ModelCommonList resourceList;
     try {
-      resourceList = mElasticsearchRepo.query(aQueryString, aFrom, aSize, aSortOrder, aFilters, aQueryContext);
+      resourceList = mESRepo.query(aQueryString, aFrom, aSize, aSortOrder, aFilters, aQueryContext,
+        checkIndices(aIndices));
     } catch (IOException e) {
       Logger.error("Could not query Elasticsearch repository", e);
       return null;
     }
     return resourceList;
   }
 
+
+  /**
+   * Concatenates the mappings of the first parameter by "AND"
+   */
+  // TODO: use this method in Controllers after merge with branch 1128-likes.
+  public ModelCommonList andQuerySqlLike(final Map<String, String> aParameters, final int aFrom, final int aSize,
```

Comment:
We might want to consider to implement this on basis of https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-match-query.html

Reply:
You mean: a sequence of method calls, replacing `" AND "`-concatenation?